repos/system_upgrade/common/actors/verifytargetversion/libraries/verifytargetversion.py

```diff
@@ -0,0 +1,54 @@
+from leapp import reporting
+from leapp.cli.commands import command_utils
+from leapp.exceptions import StopActorExecution
+from leapp.libraries.common.config import get_env
+from leapp.libraries.common.config.version import get_target_version
+from leapp.libraries.stdlib import api
+
+FMT_LIST_SEPARATOR = '\n    - '
+
+
+def process():
+    flavour = get_env('LEAPP_UPGRADE_PATH_FLAVOUR')
+    target_version = get_target_version()
+    supported_target_versions = command_utils.get_supported_target_versions(flavour)
+
+    leapp_unsupported = get_env('LEAPP_UNSUPPORTED', '0') == '1'
+    unsupported = target_version not in supported_target_versions
+
+    if unsupported and leapp_unsupported:
+        api.current_logger().warning('Upgrading to an unsupported version of the target system')
+        return
+
+    if unsupported and not leapp_unsupported:
+        hint = (
+                'Choose a supported version of the target OS for the upgrade.'
+                ' Alternatively, if you require to upgrade using an unsupported upgrade path,'
+                ' set the `LEAPP_UNSUPPORTED=1` environment variable to confirm you'
+                ' want to upgrade on your own risk.'
+        )
+
+        reporting.create_report([
+            reporting.Title('Specified version of the target system is not supported'),
+            reporting.Summary(
+                'The in-place upgrade to the specified version ({tgt_ver}) of the target system'
+                ' is not supported from the current system version. Follow the official'
+                ' documentation for up to date information about supported upgrade'
+                ' paths and future plans (see the attached link).'
+                ' The in-place upgrade is enabled to the following versions of the target system:{sep}{ver_list}'
+                .format(
+                    sep=FMT_LIST_SEPARATOR,
+                    ver_list=FMT_LIST_SEPARATOR.join(supported_target_versions),
+                    tgt_ver=target_version
+                )
+            ),
+            reporting.Groups([reporting.Groups.INHIBITOR]),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Remediation(hint=hint),
+            reporting.ExternalLink(
+                url='https://access.redhat.com/articles/4263361',
+                title='Supported in-place upgrade paths for Red Hat Enterprise Linux'
+            )
+        ])
+        raise StopActorExecution()
```

Comment:
seems my long comment about this has been dropped by GH :/ but this can be written in more simple way if initial conditions are negated.

I will push updated commit in ~15min with all changes